demo/src/example/example-page.js

```diff
@@ -8,67 +8,185 @@ import router from '../core/router';
 import { openPlayerModal } from '../player/player-dialog';
+import Drm from '../../../src/utils/Drm';
 
 /**
  * Represents a page displaying a list of dynamically loaded examples.
  *
  * @class
  */
 class ExamplePage {
+  /**
+   * The DRM vendor selection element.
+   *
+   * @private
+   * @type {Element}
+   */
+  #drmVendorEl;
+  /**
+   * The certificate url input element.
+   *
+   * @private
+   * @type {Element}
+   */
+  #certificateUrlEl;
+  /**
+   * The license url input element.
+   *
+   * @private
+   * @type {Element}
+   */
+  #licenseUrlEl;
+  /**
+   * The load bar input element.
+   *
+   * @private
+   * @type {Element}
+   */
+  #loadBarEl;
+
   /**
    * Creates an instance of ExamplePage.
    * Initializes and sets up the necessary elements and event listeners.
    */
   constructor() {
-    // Replace the content of the '.container' element with dynamically created content.
+    // Replace the content of the '.container' element with the examples.
     document.querySelector('.container').replaceChildren(...this.createContentEl());
 
-    // Add event listener for the 'click' event on the '#examples' element.
-    document.querySelector('#examples').addEventListener('click', (event) => {
+    this.#drmVendorEl = document.querySelector('#drm-vendor');
+    this.#certificateUrlEl = document.querySelector('#certificate-url-input');
+    this.#licenseUrlEl = document.querySelector('#license-url-input');
+    this.#loadBarEl = document.querySelector('#load-bar');
+
+    this.initExamples();
+    this.initLoadBar();
+  }
+
+
+  /**
+   * Initializes the examples listeners. Whenever an example button is clicked,
+   * a modal with a player is opened and the corresponding example's content is loaded.
+   */
+  initExamples() {
+    // Listen for example's click
+    document.querySelector('#examples').addEventListener('click', event => {
       const exampleEl = event.target.closest('button');
 
       // Check if the clicked element is a button and has the 'category' dataset attribute.
       if (!exampleEl || !('category' in exampleEl.parentNode.dataset)) {
         return;
       }
 
-      this.onExampleClicked(exampleEl);
+      const parent = exampleEl.parentNode;
+      const category = parent.dataset.category;
+      const exampleIdx = Array.from(parent.children)
+        .filter(child => child.tagName.toLowerCase() === 'button')
+        .indexOf(exampleEl);
+
+      openPlayerModal(Examples[category][exampleIdx]);
     });
+  }
+
+  /**
+   * Initializes the load bar element.
+   */
+  initLoadBar() {
+
+    document.querySelector('#play-content-btn')
+      .addEventListener('click', () => this.playCustomUrl());
 
-    // Add event listener for the 'keyup' event on the '#load-bar' element.
-    document.querySelector('#load-bar').addEventListener('keyup', (event) => {
+    this.#loadBarEl.addEventListener('keyup', event => {
       // Check if the 'Enter' key is pressed.
       if (event.key === 'Enter') {
```

Comment:
pls refer to https://github.com/SRGSSR/pillarbox-web/pull/96/files#r1400800857

Reply:
Got this fixed too.